int16 ball_x
int16 ball_y
